provider/application.go

```diff
@@ -0,0 +1,191 @@
+package provider
+
+import (
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/jgramoll/terraform-provider-spinnaker/client"
+)
+
+type awsProviderSettings struct {
+	UseAmiBlockDeviceMappings bool `mapstructure:"use_ami_block_device_mappings"`
+}
+
+type providerSettings struct {
+	AWS *[]*awsProviderSettings `mapstructure:"aws`
+}
+
+// Application deploy application in application
+type application struct {
+	ID                             string               `mapstructure:"id"`
+	Name                           string               `mapstructure:"name"`
+	Email                          string               `mapstructure:"email"`
+	RepoType                       string               `mapstructure:"repo_type"`
+	RepoProjectKey                 string               `mapstructure:"repo_project_key"`
+	RepoSlug                       string               `mapstructure:"repo_slug"`
+	CloudProviders                 []string             `mapstructure:"cloud_providers"`
+	ProviderSettings               *[]*providerSettings `mapstructure:"provider_settings"`
+	InstancePort                   int                  `mapstructure:"instance_port"`
+	PlatformHealthOnly             bool                 `mapstructure:"platform_health_only"`
+	PlatformHealthOnlyShowOverride bool                 `mapstructure:"platform_health_only_show_override"`
+	EnableRestartRunningExecutions bool                 `mapstructure:"enable_restart_running_executions"`
+}
+
+func (a *application) toClientApplication() *client.Application {
+	return &client.Application{
+		Name:                           a.Name,
+		Email:                          a.Email,
+		CloudProviders:                 strings.Join(a.CloudProviders, ","),
+		RepoType:                       a.RepoType,
+		RepoProjectKey:                 a.RepoProjectKey,
+		RepoSlug:                       a.RepoSlug,
+		ProviderSettings:               a.toClientProviderSettigns(a.ProviderSettings),
+		InstancePort:                   a.InstancePort,
+		PlatformHealthOnly:             a.PlatformHealthOnly,
+		PlatformHealthOnlyShowOverride: a.PlatformHealthOnlyShowOverride,
+		EnableRestartRunningExecutions: a.EnableRestartRunningExecutions,
+	}
+}
+
+func (a *application) toClientProviderSettigns(settings *[]*providerSettings) *client.ProviderSettings {
```

Comment:
I'd avoid passing a pointer to a slice -- you don't really get a whole lot from it performance wise since the compiler has a lot of optimizations for slices anyways and it usually makes the code more confusing.

Reply:
@bgphipps Just check that again and we doesn't need a pointer on slice. I follow the same structure already have in this provider. 
Example https://github.com/Fanatics/terraform-provider-spinnaker/blob/master/provider/pipeline_bake_stage.go#L22

cc/ @pviniciusfm